src/MongoDB.Bson/Serialization/Serializers/DiscriminatedInterfaceSerializer.cs

```
@@ -110,5 +110,25 @@ public override void Serialize(BsonSerializationContext context, BsonSerializati
                 _objectSerializer.Serialize(context, args, value);
             }
         }
+
+        /// <inheritdoc />
+        public bool TryGetMemberSerializationInfo(string memberName, out BsonSerializationInfo serializationInfo)
+        {
+            var memberType = _interfaceType.GetTypeInfo().GetProperty(memberName);
+            if (memberType != null)
+            {
+                var serializer = BsonSerializer.LookupSerializer(memberType.PropertyType);
```

`LookupSerializer` can potentially throw an exception which previously hadn't been thrown. However, since we already use `LookupSerializer` in line 88 (https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/79/files#diff-c7b99849615b3c9b63039543fbf1b31fR88). I assume we won't introduce any new exceptions

---

But I guess we still can create a new lookup method which won't trigger exception at all, like:
`TryLookupSerializer`.
cc: @rstam

---

> which won't trigger exception

It will allow us not to change the logic in lines like this: https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver/FieldDefinition.cs#L488

In case if `TryGetMemberSerializationInfo` doesn't find `SerializationInfo`, the condition in line `488` will be false. But in the current implementation, we will throw an exception.

---

Calling `LookupSerializer` seems suspect.

I don't think there is any guarantee that will find the correct serializer.

But maybe there is no other way?

---

As I wrote above, I guess we can create a new method: `TryLookupSerializer` which will return `false` if a serializer hasn't been found. 
As I see it's expected behavior for the `TryGetMemberSerializationInfo` method.

---

That's not what I meant.

I meant that LookupSerializer might return the *wrong* serializer.

---

>But maybe there is no other way?
In this case, it looks like this is the answer. I don't see how it can be implemented in another way.

---

If there is no way to guarantee that we find and use the correct serializer then perhaps we should not implement this.

It's better to throw an exception than to silently use the wrong serializer.

---

But don't give up yet. Let's make sure there is no way to find the correct serializer before giving up.